keras/backend/tensorflow_backend.py

```diff
@@ -1328,39 +1328,26 @@ def _cond(condition, then_lambda, else_lambda):
 
 
 def switch(condition, then_expression, else_expression):
-    '''Switches between two operations
-    depending on a scalar value (int or bool).
+    '''Switches between two operations depending on `condition`.
     Note that both `then_expression` and `else_expression`
     should be symbolic tensors of the *same shape*.
 
     # Arguments
-        condition: scalar tensor.
+        condition: a scalar, vector or tensor of the same shape with then_expression.
         then_expression: TensorFlow operation.
         else_expression: TensorFlow operation.
     '''
-    x_shape = copy.copy(then_expression.get_shape())
-    if condition.dtype != tf.bool:
-        condition = tf.cast(condition, 'bool')
-    x = _cond(condition,
-              lambda: then_expression,
-              lambda: else_expression)
-    x.set_shape(x_shape)
-    return x
-
-
-def select(condition, t, e):
-    '''Selects elements from t or e, depending on condition.
-    Note that both `t` and `e`
-    should be symbolic tensors of the *same shape*.
-
-    The condition tensor acts as a mask that chooses,
-    based on the value at each element,
-    whether the corresponding element in the output should be taken
-    from t (if true) or e (if false).
-    '''
-    if condition.dtype != tf.bool:
-        condition = tf.cast(condition, 'bool')
-    return tf.select(condition, t, e)
+    if ndim(condition) == 0:
+        x_shape = copy.copy(then_expression.get_shape())
+        if condition.dtype != tf.bool:
+            condition = tf.cast(condition, 'bool')
+        x = _cond(condition,
+                  lambda: then_expression,
+                  lambda: else_expression)
+        x.set_shape(x_shape)
+        return x
+    else:
+        return tf.select(condition, then_expression, else_expression)
```

Comment:
In your previous version you were casting the condition to `bool`, would you also want to do that here?